dpcpp/base/kernel_launch_reduction.dp.hpp

```diff
@@ -46,22 +46,18 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "dpcpp/components/reduction.dp.hpp"
 #include "dpcpp/components/thread_ids.dp.hpp"
 #include "dpcpp/components/uninitialized_array.hpp"
+#include "dpcpp/synthesizer/implementation_selection.hpp"
 
 
 namespace gko {
 namespace kernels {
 namespace dpcpp {
 
 
-using KCFG_1D = ConfigSet<11, 7>;
-constexpr auto kcfg_1d_list_simple_reduction =
-    syn::value_list<std::uint32_t, KCFG_1D::encode(512, 64),
-                    KCFG_1D::encode(512, 32), KCFG_1D::encode(512, 16),
-                    KCFG_1D::encode(256, 32), KCFG_1D::encode(256, 16),
-                    KCFG_1D::encode(256, 8)>();
+constexpr auto kcfg_1d_list_simple_reduction = kcfg_1d_list_t();
 
 
-template <std::uint32_t cfg, typename ValueType, typename KernelFunction,
+template <typename cfg, typename ValueType, typename KernelFunction,
```

Comment:
Same in this file. IMO, it should be capitalized at least, so it is clear that it is a type (`DeviceConfig` would be best IMO).